services/wallet/storage/storage.go

```diff
@@ -0,0 +1,91 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/brave-intl/bat-go/services/wallet/model"
+	"github.com/jmoiron/sqlx"
+	uuid "github.com/satori/go.uuid"
+)
+
+type Challenge struct{}
+
+func NewChallenge() *Challenge { return &Challenge{} }
+
+// Get retrieves a model.Challenge from the database by the given id.
+func (c *Challenge) Get(ctx context.Context, dbi sqlx.QueryerContext, id string) (model.Challenge, error) {
+	const q = `select * from challenge where id = $1`
+
+	var result model.Challenge
+	if err := sqlx.GetContext(ctx, dbi, &result, q, id); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return result, model.ErrNotFound
+		}
+		return result, err
+	}
+
+	return result, nil
+}
+
+// Upsert persists a model.Challenge to the database.
+func (c *Challenge) Upsert(ctx context.Context, dbi sqlx.ExecerContext, chl model.Challenge) error {
+	const q = `insert into challenge (id, created_at, nonce) values($1, $2, $3) on conflict (id) do update set created_at = $2, nonce = $3`
```

Comment:
Nit: Here and below, maybe capitalise SQL syntax?

Reply:
Going to leave this, can discus later as all the code pretty much uses lower case.